Sources/Rendering/OpenGL/Framebuffer/index.js

```diff
@@ -69,22 +69,87 @@ function vtkFramebuffer(publicAPI, model) {
     model.glFramebuffer.height = height;
   };
 
-  publicAPI.setColorBuffer = (texture, mode) => {
+  publicAPI.setColorBuffer = (texture, attachment = 0) => {
     const gl = model.context;
+
+    let glAttachment = gl.COLOR_ATTACHMENT0;
+    if (attachment > 0) {
+      if (model.openGLRenderWindow.getWebgl2()) {
+        glAttachment += attachment;
+      } else {
+        macro.vtkErrorMacro(
+          'Using multiple framebuffer attachments requires WebGL 2'
+        );
+        return;
+      }
+    }
     model.colorTexture = texture;
     gl.framebufferTexture2D(
       gl.FRAMEBUFFER,
-      gl.COLOR_ATTACHMENT0,
+      glAttachment,
       gl.TEXTURE_2D,
       texture.getHandle(),
       0
     );
   };
 
-  // publicAPI.setDepthBuffer = (texture, mode) => {
-  //   const gl = model.context;
-  //   gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.DEPTH_ATTACHMENT, gl.TEXTURE_2D, texture.getHandle(), 0);
-  // };
+  publicAPI.removeColorBuffer = (attachment = 0) => {
+    const gl = model.context;
+
+    let glAttachment = gl.COLOR_ATTACHMENT0;
+    if (attachment > 0) {
+      if (model.openGLRenderWindow.getWebgl2()) {
+        glAttachment += attachment;
+      } else {
+        macro.vtkErrorMacro(
+          'Using multiple framebuffer attachments requires WebGL 2'
+        );
+        return;
+      }
+    }
+
+    gl.framebufferTexture2D(
+      gl.FRAMEBUFFER,
+      glAttachment,
+      gl.TEXTURE_2D,
+      null,
+      0
+    );
+  };
+
+  publicAPI.setDepthBuffer = (texture, mode) => {
```

Comment:
mode -> attachment to be consistent with setColorBuffer ?

Reply:
I believe you can only attach a single depth buffer.

`mode` was unused here though, so I removed it.